### PR TITLE
adding support for directive import and merging

### PR DESCRIPTION
**Pull request by nath-abhishek**

### CODE REVIEW

**Comment by tlivings** [comment]
Looks good. Can you go ahead and add a test of this functionality against a component (instead of mocking) to cover the code you added in the GraphQLComponent class?

**Comment by nath-abhishek** [comment]
@tlivings > Looks good. Can you go ahead and add a test of this functionality against a component (instead of mocking) to cover the code you added in the GraphQLComponent class?

I have added couple of more tests. Let me know I have missed anything.

File: lib/index.js [hunk 1]
```diff
@@ -145,6 +152,18 @@ class GraphQLComponent {
   get mocks() {
     return this._mocks;
   }
+
+  get directives() {
+    return this._directives;
+  }
+
+  get importedDirectives() {
```
Comment (tlivings) [inline lib/index.js]: Does this need to be a public API?

File: lib/index.js [hunk 2]
```diff
@@ -145,6 +152,18 @@ class GraphQLComponent {
...
+    return this._importedDirectives;
+  }
+
+  get mergedDirectives() {
```
Comment (tlivings) [inline lib/index.js]: Same question — public api?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "graphql-component",
-  "version": "0.0.1-alpha.19",
+  "version": "0.0.1-alpha.20",
```
Comment (tlivings) [inline package.json]: In PRs, do not version — leave it up to maintainer to cut a new release.

File: test/test-directives.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const Test = require('tape');
+
+const GraphQLComponent = require('../lib/index');
+const Directives = require('../lib/directives');
+
+Test('test component', (t) => {
```
Comment (tlivings) [inline test/test-directives.js]: I think this test needs to validate `execute` against a component takes into account the directive, which will validate it ended up in the schema that gets created.

**Comment by tlivings** [comment]
I added more comments. I can help it land if you want by making the changes myself – just let me know.

File: lib/index.js (hunk 1 above)
Comment (nath-abhishek) [reply lib/index.js]: no, I will remove this

File: lib/index.js (hunk 2 above)
Comment (nath-abhishek) [reply lib/index.js]: yes, this one will be needed to merge directive's schema with executable schema later in server/framework code

**Comment by nath-abhishek** [comment]
@tlivings made changes as per your comments

**Pull request merged**